include/Minefield/Game/Board.h

```diff
@@ -1,56 +1,65 @@
-    struct BoardDimensions : math::Vector2UInt
-    {
-        BoardDimensions() = default;
-        BoardDimensions(unsigned int wVal, unsigned int hVal) : math::Vector2UInt{wVal, hVal} {};
-    };
+struct BoardDimensions : math::Vector2UInt
+{
+    BoardDimensions() = default;
+    BoardDimensions(unsigned int wVal, unsigned int hVal) : math::Vector2UInt{wVal, hVal} {};
+};
 
-    enum class TileType : char
-    {
-        Empty = 'o',
-        Guess = '?',
-        Bomb = '*',
-        Taken = 'x'
-    };
+enum class TileType : char
+{
+    Empty = 'o',
+    Guess = '?',
+    Bomb = '*',
+    Taken = 'x'
+};
 
-    struct Tile
-    {
-        Tile() = default;
-        Tile(TileType const& tileType, Coordinate const& coordinate) : tileType(tileType), coordinate(coordinate) {};
+struct Tile
+{
+    Tile() = default;
+    Tile(TileType const& tileType, Coordinate const& coordinate)
+    : tileType(tileType)
+    , coordinate(coordinate) {};
+
+    TileType tileType;
+    Coordinate coordinate;
+};
 
-        TileType tileType;
-        Coordinate coordinate;
-    };
+namespace utils::board
+{
+using TileMatrix = std::vector<std::vector<Tile>>;
+bool isValidCoordinate(Coordinate const& tileCoordinate, TileMatrix const& matrix, BoardDimensions const& boardDimensions);
+TileType const* getTile(Coordinate const& tileCoordinate, TileMatrix const& matrix, BoardDimensions const& boardDimensions);
+} // namespace utils::board
 
-    namespace utils::board
+class Board
+{
+public:
+    Board() = default;
+    Board(utils::board::TileMatrix matrix, BoardDimensions boardDimensions);
+
+    inline BoardDimensions const& dimensions() const noexcept
     {
-        using TileMatrix = std::vector<std::vector<Tile>>;
-        bool isValidCoordinate(Coordinate const& tileCoordinate, TileMatrix const& matrix, BoardDimensions const& boardDimensions);
-        TileType const* getTile(Coordinate const& tileCoordinate, TileMatrix const& matrix, BoardDimensions const& boardDimensions);
-    } // namespace board
-    
-    class Board
+        return mBoardDimensions;
+    }
+    inline utils::board::TileMatrix matrix() const
     {
-    public:
-        Board() = default;
-        Board(utils::board::TileMatrix matrix, BoardDimensions boardDimensions);
-
-        inline BoardDimensions const& dimensions() const noexcept { return mBoardDimensions; }
-        inline utils::board::TileMatrix matrix() const { return mMatrix; }
-        void print() const;
-        void changeTileType(Coordinate const& tileCoordinate, TileType const& tileType);
-        std::vector<Tile> getTilesOfType(TileType const& tileType) const;
-
-    private:
-        BoardDimensions mBoardDimensions;
-        utils::board::TileMatrix mMatrix;
-    };
-    
-}
\ No newline at end of file
+        return mMatrix;
+    }
+    void print() const;
+    void changeTileType(Coordinate const& tileCoordinate, TileType const& tileType);
+    std::vector<Tile> getTilesOfType(TileType const& tileType) const;
```

Comment:
#nmi is it intended to return copies of the tiles and not pointer or references to them?